foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/actions/GroupAction.java

```diff
@@ -112,30 +123,29 @@ public ActionResponse execute(GroupRequest parameter) throws QueryStoreException
                     .genFilter(parameter.getFilters()))
                     .setSearchType(SearchType.COUNT)
                     .addAggregation(rootBuilder);
+        } catch (Exception e) {
+            throw FoxtrotException.queryCreationException(parameter, e);
+        }
+        try {
             SearchResponse response = query.execute().actionGet();
             List<String> fields = parameter.getNesting();
             Aggregations aggregations = response.getAggregations();
             // Check if any aggregation is present or not
             if (aggregations == null) {
-                logger.error("Null response for Group. Request : " + parameter.toString());
                 return new GroupResponse(Collections.<String, Object>emptyMap());
             }
             return new GroupResponse(getMap(fields, aggregations));
-        } catch (QueryStoreException ex) {
-            throw ex;
-        } catch (Exception e) {
-            logger.error("Error running grouping: ", e);
-            throw new QueryStoreException(QueryStoreException.ErrorCode.QUERY_EXECUTION_ERROR,
-                    "Error running group query.", e);
+        } catch (ElasticsearchException e) {
+            throw FoxtrotException.createQueryExecutionException(parameter, e);
         }
     }
 
     private Map<String, Object> getMap(List<String> fields, Aggregations aggregations) {
         final String field = fields.get(0);
         final List<String> remainingFields = (fields.size() > 1) ? fields.subList(1, fields.size())
-                : new ArrayList<String>();
+                : new ArrayList<>();
         Terms terms = aggregations.get(Utils.sanitizeFieldForAggregation(field));
-        Map<String, Object> levelCount = new HashMap<String, Object>();
+        Map<String, Object> levelCount = new HashMap<>();
```

Comment:
This particular guava method seems redundant with Java 7 feature allowing to create map without specifying parameters. Any particular advantage of using Guava method ?